### ISSUE TITLE
Updated intrinsics have broken the provided libcore

### ISSUE THREAD

**Issue by james-darkfox** [redox-os/redox#230]
Redox cannot be compiled with the latest nightly due to changes in the intrinsics.

`$ make qemu`
https://gist.github.com/james-darkfox/7b08cf832acf44e90da0

**Comment by Ticki** [comment]
When I get time, I'll take a look and fix this.

**Comment by Chyzwar** [comment]
I have also problem with building:
![selection_002](https://cloud.githubusercontent.com/assets/5990767/10949438/c1171c8a-832a-11e5-9ff2-b3c612f454db.png)

**Comment by Ticki** [comment]
@Chyzwar: Have you updated to the latest nightly release?

**Comment by stratact** [comment]
@Chyzwar unless you're a rustc-nightly user, it's a good idea to use multirust to be able to switch between compiler versions and compile with nightly for Redox. And follow what @Ticki said.

@Ticki maybe we should state that it's required to have the latest nightly to build Redox in the README.md file?

**Comment by Chyzwar** [comment]
Thanks, It worked with multirust and nightly.

Maybe Makefile should contain a check for rustc version ?

**Comment by james-darkfox** [comment]
@Chyzwar I had that issue with 1.5.0 nightly and only the one that I've linked with the latest, 1.6.0.

**Comment by james-darkfox** [comment]
@Ticki What rustc version are you running on?

**Comment by james-darkfox** [comment]
@Ticki ^ Fixed, at least for my version.

Sidenote: I'm using the following to get VNC on `::1::5900` with qemu's console on stdio. Would anyone like me to commit this into the `Makefile`? It could get useful for debugging.
```makefile
qemu_monitor_vnc: $(BUILD)/harddrive.bin
        -qemu-system-$(ARCH) -m 1024 -usbdevice tablet -monitor stdio -hda $<
```

**Comment by Ticki** [comment]
Just updated to the latest rust nightly, it's still broken, unfortunately.

**Comment by Ticki** [comment]
Fixed by #230 (merged into branch `correctness`). Can be closed.
